File: backend-java/src/main/java/com/java_services/backend_java/api/account/domain/repositories/user/db/DeleteUserRepository.java
package com.java_services.backend_java.api.account.domain.repositories.user.db;

import org.springframework.stereotype.Repository;

import com.java_services.backend_java.api.account.domain.interfaces.Database;

@Repository
public class DeleteUserRepository {

    private final Database database;

    public DeleteUserRepository(Database database) {
        this.database = database;
    }

    public int delete(Long id) {
        if (id == null || id <= 0) {
            throw new IllegalArgumentException("O ID do usuário é obrigatório e deve ser maior que zero.");
        }
        return database.executeUpdate("DELETE FROM users WHERE id = ?", id);
    }
}

File: backend-javascript/src/account/domain/repositories/user/database/delete.user.repository.ts
import { Inject, Injectable } from "@nestjs/common";
import type { DatabaseAdapterInterface } from "@src/account/domain/interfaces/database.adapter.interface";

@Injectable()
export class DeleteUserRepository {
  constructor(@Inject('DatabaseAdapterInterface') private readonly db: DatabaseAdapterInterface) {}

  async delete(id: string): Promise<{ affectedRows: number }>  {
    return await 
    this.db.delete(
      `DELETE FROM "user" WHERE id = $1`,
     [id]
    );
  }
}